function-processor/src/test/resources/Simple.java

```diff
@@ -1,17 +1,32 @@
 package chicory.testing;
 
+import com.dylibso.chicory.function.annotations.Buffer;
+import com.dylibso.chicory.function.annotations.CString;
 import com.dylibso.chicory.function.annotations.HostModule;
 import com.dylibso.chicory.function.annotations.WasmExport;
 import com.dylibso.chicory.runtime.HostFunction;
 import com.dylibso.chicory.runtime.Memory;
 import com.dylibso.chicory.wasm.exceptions.ChicoryException;
+import java.security.SecureRandom;
 
 @HostModule("simple")
 public final class Simple {
 
     @WasmExport
-    public void print(Memory memory, int ptr, int len) {
-        System.out.println(memory.readString(ptr, len));
+    public void print(@Buffer String data) {
+        System.out.println(data);
+    }
+
+    @WasmExport
+    public void printx(@CString String data) {
+        System.out.println(data);
+    }
+
+    @WasmExport
+    public void randomGet(Memory memory, int ptr, int len) {
+        byte[] data = new byte[len];
+        new SecureRandom().nextBytes(data);
```

Comment:
nitpick: instead of generating random data we can stick with a predefined constant to avoid flakiness and make the test reproducible.

Reply:
This code is never actually called -- we only verify the generated module source code. But I'll change this to have a `Random` instance as a constructor argument, as it's likely that host modules will have instance fields in practice.